src/Compilers/VisualBasic/Portable/Symbols/MethodSignatureComparer.vb

```diff
@@ -425,8 +426,10 @@ Done:
             type1 = SubstituteType(typeSubstitution1, type1)
             type2 = SubstituteType(typeSubstitution2, type2)
 
-            If Not type1.Type.IsSameTypeIgnoringCustomModifiers(type2.Type) Then
+            If Not type1.Type.IsSameType(type2.Type, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds Or TypeCompareKind.IgnoreTupleNames) Then
                 Return SymbolComparisonResults.ReturnTypeMismatch
+            ElseIf Not type1.Type.IsSameType(type2.Type, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) Then
+                Return SymbolComparisonResults.TupleNamesMismatch
```

Comment:
>Return SymbolComparisonResults.TupleNamesMismatch [](start = 16, length = 49)

It looks like there still could be a CustomModifierMismatch in addition to TupleNamesMismatch.